libs/core/src/lib/datetime/fd-date.utils.ts

```diff
@@ -21,3 +21,34 @@ export function toIso8601(fdDate: {
         [_leftPad(fdDate.hour), _leftPad(fdDate.minute), _leftPad(fdDate.second)].join(':')
     ].join('');
 }
+
+/**
+ * @param year e.g. 2020
+ * @param month 1 = January, 12 = December
+ * @param day 1 - 31
+ * @param hour 0 - 23
+ * @param minute 0 - 59
+ * @param second 0 - 59
+ */
+// tslint:disable-next-line:max-line-length
+export function isValidByParams(params: {year: number, month?: number, day?: number, hour?: number, minute?: number, second?: number}): boolean {
+    const { year, month, day, hour, minute, second } = params;
+    if (
+        year < 1 || year > 9999
```

Comment:
Why do you find years below 1 and upper 9999 as invalid years?
With native `Date` I'm able to create such dates and `date` instance is valid.

Reply:
With these changes we are getting this:

![image](https://user-images.githubusercontent.com/3263744/124172135-52734880-da5e-11eb-9420-aa468b837a1a.png)